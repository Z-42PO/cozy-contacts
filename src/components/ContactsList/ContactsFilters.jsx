import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import { Button } from 'cozy-ui/transpiled/react'

class ContactsFilters extends Component {
  render() {
    const { setFilter, filter, allGroups, defaultFilter } = this.props

    return (
      <div className="filters-wrapper">
        <div>Sort by group : </div>
        {allGroups.map(group => (
          <Button
            key={group.id}
            label={group.name}
            theme={filter === group.id ? 'primary' : 'secondary'}
            size="small"
            onClick={() => setFilter(group.id)}
          />
        ))}
        <Button
          label="All"
          size="small"
          theme={filter === defaultFilter ? 'primary' : 'secondary'}
          onClick={() => setFilter(defaultFilter)}
        />
      </div>
    )
  }
}

ContactsFilters.propTypes = {
  setFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
  allGroups: PropTypes.array.isRequired,
  defaultFilter: PropTypes.string.isRequired
}

export default ContactsFilters
