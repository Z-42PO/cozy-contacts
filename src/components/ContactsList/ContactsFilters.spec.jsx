import React from 'react'

import AppLike from '../../tests/Applike'
import ContactsFilters from './ContactsFilters'
import renderer from 'react-test-renderer'

describe('ContactsFilters', () => {
  const allGroups = [
    { id: 'group_id_family', name: 'Family' },
    { id: 'group_id_work', name: 'Work' }
  ]
  const defaultFilter = 'all'

  it('should display the filters with default filter selected', () => {
    const instance = (
      <AppLike>
        <ContactsFilters
          allGroups={allGroups}
          filter={defaultFilter}
          defaultFilter={defaultFilter}
          setFilter={() => {}}
        />
      </AppLike>
    )
    const tree = renderer.create(instance).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should display the filters with "Family" selected', () => {
    const instance = (
      <AppLike>
        <ContactsFilters
          allGroups={allGroups}
          filter={allGroups[0].id}
          defaultFilter={defaultFilter}
          setFilter={() => {}}
        />
      </AppLike>
    )
    const tree = renderer.create(instance).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should display the filters with "Work" selected', () => {
    const instance = (
      <AppLike>
        <ContactsFilters
          allGroups={allGroups}
          filter={allGroups[1].id}
          defaultFilter={defaultFilter}
          setFilter={() => {}}
        />
      </AppLike>
    )
    const tree = renderer.create(instance).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
