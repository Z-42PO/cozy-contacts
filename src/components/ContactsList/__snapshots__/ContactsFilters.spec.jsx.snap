// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsFilters should display the filters with "Family" selected 1`] = `
<div
  className="filters-wrapper"
>
  <div>
    Sort by group : 
  </div>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        Family
      </span>
    </span>
  </button>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        Work
      </span>
    </span>
  </button>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        All
      </span>
    </span>
  </button>
</div>
`;

exports[`ContactsFilters should display the filters with "Work" selected 1`] = `
<div
  className="filters-wrapper"
>
  <div>
    Sort by group : 
  </div>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        Family
      </span>
    </span>
  </button>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        Work
      </span>
    </span>
  </button>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        All
      </span>
    </span>
  </button>
</div>
`;

exports[`ContactsFilters should display the filters with default filter selected 1`] = `
<div
  className="filters-wrapper"
>
  <div>
    Sort by group : 
  </div>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        Family
      </span>
    </span>
  </button>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        Work
      </span>
    </span>
  </button>
  <button
    className="styles__c-btn___-2Vnj styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh"
    onClick={[Function]}
    title={null}
    type="submit"
  >
    <span>
      <span
        className={null}
      >
        All
      </span>
    </span>
  </button>
</div>
`;
